cores/esp8266/umm_malloc/umm_malloc_cfg.h

```diff
@@ -155,12 +377,66 @@ extern char _heap_start[];
  * 4 bytes, so there might be some trailing "extra" bytes which are not checked
  * for corruption.
  */
+
+/*
+ * Not normally enabled. Full intergity check may exceed 10us.
+ */
 /*
 #define UMM_INTEGRITY_CHECK
-*/
+ */
+
+#ifdef UMM_INTEGRITY_CHECK
+   int umm_integrity_check( void );
+#  define INTEGRITY_CHECK() umm_integrity_check()
+   extern void umm_corruption(void);
+#  define UMM_HEAP_CORRUPTION_CB() DBGLOG_FUNCTION( "Heap Corruption!" )
+#else
+#  define INTEGRITY_CHECK() 0
+#endif
+
+
+/////////////////////////////////////////////////
+
+#if defined(DEBUG_ESP_PORT) || defined(DEBUG_ESP_CORE)
+#define UMM_POISON_CHECK
+
+// less agressive checking, checks at realloc and free only.
+#define UMM_POISON_CHECK_LITE
+// Note, UMM_POISON is deprecated
+#define UMM_POISON
+#endif
```

Comment:
I have changed the default for UMM_POISON(_CHECK) to not do a comprehensive full heap check on every realloc and free. Instead, it only checks the allocation. Hmm, no hard notes here, I believe it was taking maybe a couple hundred us to do the whole heap. It really varies with the number of entries in the heap.

An alternative I thought of to just checking the current allocation would be to expand the check to its immediate neighbors.

Reply:
Checking the current + neighbors in all three cases sounds like a good idea to me! Minimum overhead for reasonable sanity check.

Reply:
For `realloc()` and `free()`, expanded the poison check on the current allocation to include its nearest allocated neighbors in the heap. That is to say, if the nearest neighbor is free, then the next one over is checked. And this is done in both directions. 

For `malloc()` added a check for the neighbors of the newly selected allocation before its use.